src/vos/tests/vts_checksum.c

```diff
@@ -61,7 +61,7 @@ struct test_case_args {
 };
 
 #define	CSUM_FOR_ARRAYS_TEST_CASE(state, ...) \
-	csum_for_arrays_test_case(state, (struct test_case_args)__VA_ARGS__)
+	csum_for_arrays_test_case(state, &(struct test_case_args)__VA_ARGS__)
```

Comment:
Why the __VA_ARGS__ and not just args since it's always one parameter.

I didn't know the compiler would let you do this.  Interestingly, if the argument were an integer, it would not let you cast it to a constant but generates an error.

I think it might be a little clearer to do something like

do { \
   struct test_case_args __args = args; \
   csum_for_arrays_test_suite(state, &__args); \
} while (0)

Same effect but a little more explicit.

Reply:
Updated.